content/articles/data-access-with-ado-dotnet-in-micro-focus-visual-cobol/index.md

```diff
@@ -0,0 +1,217 @@
+
+### Create a Connection
+To use a database, we must first establish a connection. If you've ever used ODBC with embedded SQL, you'll know that you must first establish the connection in the Control Panel's ODBC administrators panel before issuing the SQL connect query. To openControl Panel's ODBC click Administrative Tools in the Control Panel. Click Data Sources under Administrative Tools (ODBC). In Windows Server 2008, open the ODBC Data Source Administrator. Select Administrative Tools from the Start menu, then Data Sources (ODBC). "EXEC SQL CONNECT TO DATABASE END-EXEC" was comparable to the connect statement.
+
+The connection in ADO is significantly more straightforward. To establish a database connection, the SQLConnection object is needed. One of the advantages of ADO is that we no longer need to configure the connection in the ODBC administration panel, which simplifies the configuration and distribution. Instead, we just construct our database connection string, create a database connection, and call the `Open` method on the new object. The connection is formed in our example using the following code:
+
+```sql
+set sqlConnection to new type System.Data.SqlClient.SqlConnection.
+set sqlConnection::ConnectionString to
+"Data Source=WIN8VC23\SQL12DEV;Initial Catalog-Northwind;User ID-sa;Password-NFUSER;MultipleActiveResultsSets=True".
+invoke sqlConnection::Open().
+```
+
+From the above code, the statement creates a `NEW` SQLConnection object with the name sqlConnection. Then the `connection string attribute` is then used to give parameters for the connection. The `Open` method of sqlConnection is then used to establish the connection in the following statement. The connection string is the most difficult component of this operation. We've specified the User ID, password, and database to which we're attempting to connect. Workspace variables that contain the connection string can be created and passed to methods for setting up connections, or they can be created directly in the connection-starting procedure. The connection string can be defined in a variety of ways. Only one example of syntax can be shown.
+
+### Create a Command
+Now that we've created our connection, we need to inform the database of what we're trying to accomplish. A command structure is used to do this. If you've ever used embedded SQL, you'll know what I'm talking about. Your prior statement would have been `EXEC SQL SELECT * FROM EMPLOYEES END-EXEC.` We're going to pull some data from the Employee database and present it in the console window. The following statements are used to establish the command:
```

Comment:
Avoid such statements: `If you've ever used embedded SQL, you'll know what I'm talking about.`

Reply:
Where do we have the `Employee` database?